prov/psm/src/psmx_init.c

```diff
@@ -475,10 +485,19 @@ PSM_INI
 	int check_version;
 	int err;
 
-	psmx_env.name_server	= psmx_get_int_env("OFI_PSM_NAME_SERVER", 1);
-	psmx_env.am_msg		= psmx_get_int_env("OFI_PSM_AM_MSG", 0);
-	psmx_env.tagged_rma	= psmx_get_int_env("OFI_PSM_TAGGED_RMA", 1);
-	psmx_env.uuid		= getenv("OFI_PSM_UUID");
+	psmx_env.name_server =
+		psmx_get_int_env("name_server",
+				"Whether there is a PSM name server or not",
+				1);
+	psmx_env.am_msg =
+		psmx_get_int_env("am_msg",
+				"Whether PSM supports active messages or not",
+				0);
+	psmx_env.tagged_rma =
+		psmx_get_int_env("tagged_rma",
+				"Whether PSM supports tagged RMA or not",
+				1);
+	psmx_reg_and_read("uuid", "Unique ID for PSM fabric", &psmx_env.uuid);
```

Comment:
@j-xiong - This is separate from this patch, but can you look at moving reading these values until immediately before they are needed?